anthos-bm-gcp-bash/install_admin_cluster.sh

```diff
@@ -242,88 +285,108 @@ EOF
 # [END anthos_bm_gcp_bash_admin_add_ssh_keys]
 printf "✅ Successfully set up SSH access from admin workstation to cluster node VMs.\n\n"
 
-# initiate Anthos on bare metal installation from the admin workstation
-printf "🔄 Installing Anthos on bare metal...\n"
-# [START anthos_bm_gcp_bash_admin_install_abm]
-gcloud compute ssh root@$VM_WS --zone "${ZONE}" <<EOF
-set -x
-export PROJECT_ID=\$(gcloud config get-value project)
-ADMIN_CLUSTER_NAME=\$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/cluster_id -H "Metadata-Flavor: Google")
-BMCTL_VERSION=\$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/bmctl_version -H "Metadata-Flavor: Google")
-export ADMIN_CLUSTER_NAME
-export BMCTL_VERSION
-bmctl create config -c \$ADMIN_CLUSTER_NAME
-cat > bmctl-workspace/\$ADMIN_CLUSTER_NAME/\$ADMIN_CLUSTER_NAME.yaml << EOB
----
-gcrKeyPath: /root/bm-gcr.json
-sshPrivateKeyPath: /root/.ssh/id_rsa
-gkeConnectAgentServiceAccountKeyPath: /root/bm-gcr.json
-gkeConnectRegisterServiceAccountKeyPath: /root/bm-gcr.json
-cloudOperationsServiceAccountKeyPath: /root/bm-gcr.json
----
-apiVersion: v1
-kind: Namespace
-metadata:
-  name: cluster-\$ADMIN_CLUSTER_NAME
----
-apiVersion: baremetal.cluster.gke.io/v1
-kind: Cluster
-metadata:
-  name: \$ADMIN_CLUSTER_NAME
-  namespace: cluster-\$ADMIN_CLUSTER_NAME
-spec:
-  type: admin
-  anthosBareMetalVersion: \$BMCTL_VERSION
-  gkeConnect:
-    projectID: \$PROJECT_ID
-  controlPlane:
-    nodePoolSpec:
-      clusterName: \$ADMIN_CLUSTER_NAME
-      nodes:
-      - address: 10.200.0.3
-  clusterNetwork:
-    pods:
-      cidrBlocks:
-      - 192.168.0.0/16
-    services:
-      cidrBlocks:
-      - 10.96.0.0/20
-  loadBalancer:
-    mode: bundled
-    ports:
-      controlPlaneLBPort: 443
-    vips:
-      controlPlaneVIP: 10.200.0.48
-  clusterOperations:
-    # might need to be this location
-    location: us-central1
-    projectID: \$PROJECT_ID
-  storage:
-    lvpNodeMounts:
-      path: /mnt/localpv-disk
-      storageClassName: node-disk
-    lvpShare:
-      numPVUnderSharedPath: 5
-      path: /mnt/localpv-share
-      storageClassName: local-shared
-  nodeConfig:
-    podDensity:
-      maxPodsPerNode: 250
-EOB
-
-bmctl create cluster -c \$ADMIN_CLUSTER_NAME
+
+if [[ "$OPT" = "$RUN_ALL" ]];
+then
```

Comment:
As we discussed in chat, indenting this section is causing an error:
-bash: line 66: warning: here-document at line 8 delimited by end-of-file (wanted `EOB')

I found this Stackoverflow https://stackoverflow.com/questions/18660798/here-document-gives-unexpected-end-of-file-error that has quite a few suggestions. You can investigate as to what is causing the error. Alternatively, you can switch the logic so you don't have to indent. 

After the script finishes setting up the GCE VMs, do this:

if [[ "$OPT" = "$SETUP_ONLY" ]];
then
  # [START anthos_bm_gcp_bash_admin_gce_info_setup_only]
  printf "✅ GCE Infrastructure setup complete. Please check the logs for any errors!!!\n\n"
  printf "✅ If you do not see any errors in the output log, then you now have the following setup:\n\n"
  printf "|---------------------------------------------------------------------------------------------------------|\n"
  printf "| VM Name               | L2 Network IP (VxLAN) | INFO                                                    |\n"
  printf "|---------------------------------------------------------------------------------------------------------|\n"
  printf "| abm-admin-cluster-cp  | 10.200.0.3            | 🌟 Ready for use as control plane for the admin cluster |\n"
  printf "| abm-user-cluster-cp   | 10.200.0.4            | 🌟 Ready for use as control plane for the user cluster  |\n"
  printf "| abm-user-cluster-w1   | 10.200.0.5            | 🌟 Ready for use as worker for the user cluster         |\n"
  printf "| abm-user-cluster-w2   | 10.200.0.6            | 🌟 Ready for use as worker for the user cluster         |\n"
  printf "|---------------------------------------------------------------------------------------------------------|\n\n"
  # [END anthos_bm_gcp_bash_admin_gce_info_setup_only]
  printf "Exiting...\n"
  exit 0
fi  

The script exits if you select Setup only. The script continues otherwise.

Reply:
I have made some changes to fix this issue. PTAL and see if it works now.